src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -456,34 +457,38 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     const healthPerNamespace = new Map<String, Promise<NamespaceHealth>>();
     // Asynchronously fetch health
     cy.nodes().forEach(ele => {
-      const fqService = ele.data('service');
-      if (fqService && (ele.data('isGroup') || !ele.data('parent'))) {
-        const serviceParts = fqService.split('.');
-        if (serviceParts.length < 2) {
-          // Ignore health for special nodes such as "unknown"
-          return;
-        }
-        const service = serviceParts[0];
-        const namespace = serviceParts[1];
+      const namespace = ele.data('namespace');
+      const app = ele.data('app');
+      const workload = ele.data('workload');
+
+      if ((workload !== 'unknown' || app !== 'unknown') && (ele.data('isGroup') || !ele.data('parent'))) {
```

Comment:
It looks correct but I think it's still need some changes later on. `API.getNamespaceHealth` will have to get a parameter (or be renamed) so that we know if we're fetching app-health, service-health or workload-health for the whole namespace. But that will come later.